src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -403,6 +404,11 @@ private ComponentState GetOptionalComponentState(int componentId)
         /// </summary>
         protected virtual void ProcessPendingRender()
         {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(nameof(Renderer), "Cannot process pending renders after the renderer has been disposed.");
+            }
+
```

Comment:
Should we log something here and no-op instead? If I recall correctly we no-op in other cases. I'm concerned about getting an unhandled exception thrown from a weird place.

Reply:
I think no-oping is safer as there might be legitimate scenarios where something got queued into the sync context and the context got disposed before it got a chance to execute (as things get serialized).

I think we need to at least check on the RemoteRenderer before we call ProcessPendingRender during OnRenderCompleted

Reply:
> I think we need to at least check on the RemoteRenderer before we call ProcessPendingRender during OnRenderCompleted

Great point. Yes. I'm adding a check for that.

> Should we log something here and no-op instead?

Since we have no reason to believe it's at possible to make this happen through normal coding patterns, I'd be more interested in having a way of getting alerted if we're wrong and it does somehow happen.

If we can articulate definite scenarios where we want to support this I'm totally open to changing this though!

> I think no-oping is safer as there might be legitimate scenarios where something got queued into the sync context and the context got disposed before it got a chance to execute (as things get serialized).

Do you have specific scenarios in mind? The ones I'm aware of are:

 * If a component asks to be rendered - this already no-opped if the renderer (or component) was already disposed.
 * When adding a new root component. If the renderer was already disposed, I think it's good to throw, since it's so bizarre to be adding a new root component then. For normal server-side Blazor, the registered root component will be added immediately during host creation, so definitely before disposal.
 * When resuming the queue processing after an ACK, which will now be no-opped by the updated logic here.

Reply:
I was a bit worried about the consequences of missing out a scenario more than anything and the potential consequences of it. Just wanted to be on the safer side.